lib/connection-manager.js

```diff
@@ -1,125 +1,90 @@
 'use strict'
 
 // External Modules
-const Agent = require('agentkeepalive')
 const async = require('async')
 const debug = require('debug')('logdna:lib:connection-manager')
 const request = require('request')
-const url = require('url')
 
 // Internal Modules
-const fileUtils = require('./file-utilities')
-const getProxyFromURI = require('./getProxyFromURI')
-const linebuffer = require('./linebuffer')
-const utils = require('./utils')
-
-// Variables
-var firstrun
+const client = require('./logger-client.js')
+const fileUtils = require('./file-utilities.js')
+const utils = require('./utils.js')
 
 module.exports = {
-  connectLogServer
+  checkHealth
+, setupLoggerThenStart
 }
 
 // Send POST Request to HealthCheck Endpoint
-// TODO(darinspivey) This should have a callback so that we know when it's done.  Currently, we have to `sleep`
-function checkHealth(config) {
-  if (config.req) {
-    return async.retry({
-      times: 3
-    , interval: (retryCount) => {
-        return 50 * Math.pow(2, retryCount)
+function checkHealth(config, cb) {
+  async.retry({
+    times: 3
+  , interval: (retryCount) => {
+      return 50 * Math.pow(2, retryCount)
+    }
+  , errorFilter: (error) => {
+      return config.DEFAULT_HTTP_ERRORS && config.DEFAULT_HTTP_ERRORS.includes(error) || error === 'INTERVAL_SERVER_ERROR'
+    }
+  }, (reqCallback) => {
+    request.post({
+      url: utils.buildUrl(config.LOGDNA_LOGHOST, config.LOGDNA_LOGPORT, config.LOGDNA_LOGSSL, config.LOGDNA_HEALTHCHECK)
+    , headers: {
+        'Content-Type': 'application/json; charset=UTF-8'
+      , ...config.UserAgent
       }
-    , errorFilter: (error) => {
-        return config.DEFAULT_HTTP_ERRORS && config.DEFAULT_HTTP_ERRORS.includes(error) || error === 'INTERVAL_SERVER_ERROR'
+    , auth: {
+        user: 'x'
+      , pass: config.key
       }
-    }, (reqCallback) => {
-      return config.req({
-        url: utils.buildUrl(config.LOGDNA_LOGHOST, config.LOGDNA_LOGPORT, config.LOGDNA_LOGSSL, config.LOGDNA_HEALTHCHECK)
-      , body: JSON.stringify({
-          e: 's'
-        , m: process.memoryUsage()
-        })
-      , qs: {
-          timestamp: Date.now()
-        }
-      }, (error, response, body) => {
-        if (error) return reqCallback(error.code)
-        if (response && response.statusCode >= 500) return reqCallback('INTERVAL_SERVER_ERROR')
-        return reqCallback(null, {...JSON.parse(body || '{}'), statusCode: response.statusCode})
+    , body: JSON.stringify({
+        e: 's'
+      , m: process.memoryUsage()
```

Comment:
do we need to continue sending this with the removal of restarts? I'm not sure of the cost of `process.memoryUsage()`

Reply:
yea, it seems like we don't do anything with that info - it was needed for `autoRestart` but now we are getting rid of that, no?

Reply:
I'll remove!